### PR TITLE
feat: new pattern `avm/ptn/security/security-center`

### LINKED ISSUE #802: [Module Proposal]: `avm/ptn/security/security-center`
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Bicep

### Module Classification?

Resource Module

### Module Name

avm/res/security/security-center

### Module Details

Azure Security Center

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

tony-box

### (Optional) Secondary Module Owner's GitHub Username (handle)

_No response_

### PR DESCRIPTION
**Pull request by tony-box**
## Description

Migration of ptn/security-center from CARML -> AVM.

https://github.com/Azure/Azure-Verified-Modules/issues/802

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.ptn.security.security-center](https://github.com/tony-box/bicep-registry-modules/actions/workflows/avm.ptn.security.security-center.yml) |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/ptn/security/security-center/tests/e2e/max/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      scope: '/subscriptions/${subscription().subscriptionId}'
```
Comment (AlexanderSehr) [inline avm/ptn/security/security-center/tests/e2e/max/main.test.bicep:52]:
```suggestion
      scope: subscription().Id
```

File: avm/res/api-management/service/README.md
Comment (AlexanderSehr) [inline avm/res/api-management/service/README.md:1]: Unintended change?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -5,6 +5,7 @@
 /avm/utilities/ @Azure/avm-core-team-technical-bicep
 /avm/ptn/authorization/policy-assignment/ @Azure/avm-ptn-authorization-policyassignment-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/ptn/authorization/role-assignment/ @Azure/avm-ptn-authorization-roleassignment-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/ptn/security/security-center/ @Azure/avm-ptn-security-securitycenter-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (AlexanderSehr) [inline .github/CODEOWNERS:8]: Soft-block until team is confirmed / files shows no more errors

File: avm/res/api-management/service/README.md
Comment (tony-box) [reply avm/res/api-management/service/README.md:1]: not sure why this is here. I don't see it in any of the latest commits.

Comment (tony-box) [reply avm/res/api-management/service/README.md:1]: @AlexanderSehr please re-review. Latest code has the correct text

File: .github/CODEOWNERS (hunk 2 above)
Comment (tony-box) [reply .github/CODEOWNERS:8]: @AlexanderSehr  please review again. I have added the appropriate groups

**Review by AlexanderSehr — DISMISSED** [review]

File: avm/res/api-management/service/README.md
Comment (AlexanderSehr) [reply avm/res/api-management/service/README.md:1]: Still is very much there. To resolve this, I'd recomment to just pick the latest from upstream main, overwrite the content in this file in your branch and push

**Review by tony-box — COMMENTED** [review]
Latest is working

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**